test/test_autograd.py

```diff
@@ -3022,18 +3022,19 @@ def test_igammac(self):
         gradgradcheck(torch.igamma, (s, x))
 
     def test_chain_matmul(self):
-        def gen_matrices(p):
+        def gen_matrices(p, dtype):
             matrices = []
             for (pi, pi_1) in zip(p[:-1], p[1:]):
-                matrices.append(torch.randn(pi, pi_1).requires_grad_())
+                matrices.append(torch.randn(pi, pi_1, dtype=dtype).requires_grad_())
```

Comment:
How is randn defined on tensors of dtype complex? (Just checking that it does output a complex-valued tensor, rather than a real-valued complex dtype tensor)